functions/nfts/asset/[[index]].ts

```diff
@@ -6,22 +6,22 @@ export const onRequest: PagesFunction = async ({ params, request, next }) => {
   const { index } = params
   const collectionAddress = String(index[0])
   const tokenId = String(index[1])
+  const cache = caches.default
   const assetPromise = getAsset(collectionAddress, tokenId, request.url)
   const resPromise = next()
+  const cachePromise = cache.match(request.url)
   try {
-    const cache = caches.default
-    const response = await cache.match(request.url)
-    if (response) {
-      const data = JSON.parse(await response.text())
-      return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(await resPromise)
+    const [graphData, cacheResponse, res] = await Promise.all([assetPromise, cachePromise, resPromise])
```

Comment:
This will always wait for `graphData` to resolve, even if it is already cached.

Instead, you should only call `graphData` if it is not yet cached. You could construct a helper utility (that you could use across all of these files) that - given a url and getter, tries the cache and then the network:

```
async function matchThenFetch<T>(url: string, fetcher: () => T) {
  const match = await caches.default.match(url)
  return match ? match : fetcher()
}
```

Reply:
Could I also just take it out of the parallelization?